### DISCUSSION TITLE
How do I pass a `vector<bool>` as an argument in my frontend

### DISCUSSION THREAD

**Discussion by Pelumi527** [aptos-labs/aptos-developer-discussions#124] [Questions] [answered]
I am trying to pass a vector<bool> as args to a function I am trying to call? tried passing it in array but still getting an error.

**Comment by 0x-j** [comment, marked as answer]
Hi checkout this example I made, [here's the frontend code](https://github.com/0xaptosj/playground/blob/main/src/components/Run.tsx#L17), calling a [function](https://github.com/0xaptosj/playground/blob/main/sources/playground_1.move#L9) which takes a `vector<bool>`, the contract doesn't do anything it just emits an event. Here's a [tx](https://explorer.aptoslabs.com/txn/999208420?network=testnet) calling it.

**Comment by Pelumi527** [comment]
@0xaptosj Oh thanks, so after putting it in square bracket it giving me `generic error`.
so for context I am trying to call the aptos module function `create_collection_script` and am trying to pass the `mutate_setting` argument
```
const result = await signAndSubmitTransaction({
        sender: account?.address,
        data: {
          function: "0x3::token::create_collection_script",
          typeArguments: [],
          functionArguments: [
            collectionName,
            description,
            uri,
            maximumSupply,
            [true,true],
          ],
        },
      });
```
so here is what I am passing

The entry function 
```
public entry fun [create_collection_script](https://aptos.dev/reference/move/token.md#0x3_token_create_collection_script)(
    creator: &signer,
    name: String,
    description: String,
    uri: String,
    maximum: u64,
    mutate_setting: vector<bool>,
```

**Reply by 0x-j** [reply]
Ok turns out mutate_setting needs 3 bools instead of 2. This is the [source code](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token.move#L1254-L1260) reading that vector. We really should have better error message instead of showing generic error.
